src/kvstore/RocksEngine.cpp

```diff
@@ -425,5 +425,66 @@ ResultCode RocksEngine::flush() {
     }
 }
 
+ResultCode RocksEngine::createCheckpoint(const std::string& name) {
+    LOG(INFO) << "Begin checkpoint : " << dataPath_;
+
+    /*
+     * The default checkpoint directory structure is :
+     *   |--FLAGS_data_path
+     *   |----nebula
+     *   |------space1
+     *   |--------data
+     *   |--------wal
+     *   |--------checkpoints
+     *   |----------snapshot1
+     *   |------------data
+     *   |------------wal
+     *   |----------snapshot2
+     *   |----------snapshot3
+     *
+     */
+
+    auto checkpointPath = folly::stringPrintf("%s/checkpoints/%s/data",
+                                              dataPath_.c_str(), name.c_str());
+    LOG(INFO) << "Target checkpoint path : " << checkpointPath;
+    if (fs::FileUtils::exist(checkpointPath)) {
+        LOG(ERROR) << "The snapshot file already exists: " << checkpointPath;
+        return ResultCode::ERR_CHECKPOINT_ERROR;
+    }
+
+    auto parent = checkpointPath.substr(0, checkpointPath.rfind('/'));
+    if (!FileUtils::exist(parent)) {
+        FileUtils::makeDir(parent);
+    }
+
+    rocksdb::Checkpoint* checkpoint;
+    rocksdb::Status status = rocksdb::Checkpoint::Create(db_.get(), &checkpoint);
+    std::unique_ptr<rocksdb::Checkpoint> cp(checkpoint);
+    if (!status.ok()) {
+        LOG(ERROR) << "Init checkpoint Failed: " << status.ToString();
+        return ResultCode::ERR_CHECKPOINT_ERROR;
+    }
+    status = cp->CreateCheckpoint(checkpointPath, 0);
+    if (!status.ok()) {
+        LOG(ERROR) << "Create checkpoint Failed: " << status.ToString();
+        return ResultCode::ERR_CHECKPOINT_ERROR;
+    }
+    return ResultCode::SUCCEEDED;
+}
+
+ResultCode RocksEngine::dropCheckpoint(const std::string& name) {
+    auto checkpointPath = folly::stringPrintf("%s/checkpoints/%s",
```

Comment:
I think we'd better move the logic outside. 
Because it not only remove the snapshot dir but also the wal dirs.  wdyt?